jupyter-extension/jupyterlab_pachyderm/mount_server_client.py

```diff
@@ -92,6 +101,24 @@ async def _ensure_mount_server(self):
                     if tries == 10:
                         get_logger().debug("Unable to start mount server...")
                         return False
+                
+                if self.nopriv:
+                    # Using un-shareed mount, replace /pfs with a softlink to the mount point
+                    mount_server_proc = subprocess.run(['pgrep','mount-server'], capture_output=True)
+                    mount_server_pid = mount_server_proc.stdout.decode("utf-8")
+                    if not mount_server_pid or not int(mount_server_pid) > 0:
+                        get_logger().debug(f"Unable to find mount-server process: {mount_server_pid}")
+                        return False
+                    mount_server_pid = int(mount_server_pid)
+                    get_logger().info(f"Link non-privileged /pfs to /proc/{mount_server_pid}/root/mnt{self.mount_dir}")
```

Comment:
unshare runs the specified program in a new namespace -- https://man7.org/linux/man-pages/man1/unshare.1.html

One cannot perform a mount except as root even in a container.   Outside of a container fusermount works because it is a setuid program, however that doesn't work in a container either.   Using ushare allows the mount in a private namespace that no-other program can see  -- since it is isolated and cannot impact any other part of the system the mount is allowed.     The mount is not directly visible though, so for other programs to see the result except through the /proc filesystem which is link into /pfs to make visible.

See the internal [HPE design note](https://rndwiki-pro.its.hpecorp.net/pages/viewpage.action?pageId=214902618) on this  area.   

When put into NONPRIV_CONTAINER mode, the mount-server does not access /pfs at all.  Instead it mounts to /mnt/pfs and then the plugin (which started the mount-server, but is outside the private namespace) can setup a link to the /mnt/pfs that the mount-server sees.